src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashBuildTrigger.java

```
@@ -309,11 +317,22 @@ private boolean hasCauseFromTheSamePullRequest(@Nullable List<Cause> causes, @Nu
 
     @Override
     public void run() {
-        if(!this.getBuilder().getJob().isBuildable()) {
+        if (!this.getBuilder().getJob().isBuildable()) {
             logger.info(format("Build Skip (%s).", getBuilder().getJob().getName()));
-        } else {
+        } else if(checkAlreadyQueued.compareAndSet(false, true)) {
             logger.info(format("Build started (%s).", getBuilder().getJob().getName()));
-            this.stashPullRequestsBuilder.run();
+            descriptor.getExecutorService().execute(new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        stashPullRequestsBuilder.run();
```

Is it by intentention that all exceptions are ignored now?

---

Let's say it is not intentional. Let me add the fix for this.